bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/rewriter/gemini/GeminiRewriter.java

```diff
@@ -33,29 +33,32 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class GeminiRewriter {
 
     private static final Logger logger = LoggerFactory.getLogger(GeminiRewriter.class);
-    private static final String API_URI_TEMPLATE = "https://us-central1-aiplatform.googleapis.com/v1/projects/%s/locations/us-central1/publishers/google/models/gemini-1.0-pro:streamGenerateContent?alt=sse";
-
-    public static void rewriteSQL(InputQuery inputQuery, List<AntiPatternVisitor> visitorsThatFoundAntiPatterns,
-                                  String projectId, PromptYamlReader promptYamlReader) throws IOException {
-        try {
-            String queryStr = inputQuery.getQuery();
-            for (AntiPatternVisitor visitor : visitorsThatFoundAntiPatterns) {
-                String prompt = promptYamlReader.getAntiPatternNameToPrompt().get(visitor.getName());
-                if (prompt != null) {
-                    prompt = String.format(prompt, queryStr).replace("%%","%");
-                    queryStr = processPrompt(prompt, projectId);
-                }
-            }
-            if (!queryStr.equals(inputQuery.getQuery())) {
-                inputQuery.setOptimizedQuery(queryStr);
+    private static final String API_URI_TEMPLATE = "https://us-central1-aiplatform.googleapis.com/v1/projects/%s/locations/us-central1/publishers/google/models/gemini-1.5-pro-preview-0409:generateContent?alt=sse";
+
+
+    public static void rewriteSQL(InputQuery inputQuery,
+                                  List<AntiPatternVisitor> visitorsThatFoundAntiPatterns,
+                                  AntiPatternHelper antiPatternHelper) throws IOException {
+
+        String queryStr = inputQuery.getQuery();
+        QueryVisitorRewriter queryVisitorRewriter = new QueryVisitorRewriter(antiPatternHelper);
+        for (AntiPatternVisitor visitor : visitorsThatFoundAntiPatterns) {
+            try {
+                queryStr = queryVisitorRewriter.rewriteSQL(queryStr, visitor, 3);
```

Comment:
to be consistent could we rename visitor as `visitorThatFoundAntiPattern`. I really like the `visitorsThatFoundAntiPatterns` naming, it makes the code easier to read, but let's be consistent.

Reply:
Done